### PR TITLE
Allow to ignore specific warnings

### LINKED ISSUE #313: CLI option to ignore alternate locations in PDB
Currently, if there are alternate locations in a PDB input, then martinize issues a warning for each alternate location it encounters that is not `A` or empty. This means that if a PDB file uses alternate locations, then many warnings are issued, the number of which is only predictable by analysing the input file. These warnings are difficult to deal with in scripts because ignoring just these warnings means one needs to analyse the input files or run martinize twice to know how many warnings to ignore. This can easily lead to `-maxwarn 9999` being common in scripts.

A solution would be an option to choose the alternate, but this option is unlikely to be used. A simpler option is one that ignores alternates other than `A` or empty. This is the behaviour the most likely to be useful. I suggest `-ignore-alternate` with a default to `False`.

### PR DESCRIPTION
**Pull request by jbarnoud**
Fixes #313 by implementing the way of ignoring specific warning types discussed in the issue.

### CODE REVIEW

File: bin/martinize2 [hunk 1]
```diff
@@ -269,6 +270,31 @@ def _cys_argument(value):
         return result
 
 
+def maxwarn(value):
+    msg = ("Values for the -maxwarn option must be the name of a "
+           "warning type, a number, or following the format "
+           "'<warning-type>:<count>' where <warning-type> if the name "
```
Comment (pckroon) [inline bin/martinize2:276]: s/if/is/

File: vermouth/log_helpers.py [hunk 2]
```diff
@@ -241,3 +240,39 @@ def get_logger(name):
...
+                deduct_all.add(warning_type)
+            else:
+                specs[warning_type] = max(specs.get(warning_type, 0), count)
+    blanket_ignore = specs.get(None, 0)
+    warning_count = counter.counts[level]
+    total = number_of_warnings
+    for warning_type, count in warning_count.items():
+        type_count = warning_count[warning_type]
+        if warning_type in specs:
+            total -= max(0, min(count, specs[warning_type]))
```
Comment (pckroon) [inline vermouth/log_helpers.py:272]: Could use a comment

File: bin/martinize2 [hunk 3]
```diff
@@ -269,6 +270,31 @@ def _cys_argument(value):
...
+           "of the warning type to ignore, and <count> is the number of "
+           "warning of that type to ignore. "
+           "'{value}' is not a valid value.".format(value=value))
+    splitted = value.split(':')
+    if len(splitted) == 1:
+        try:
+            count = int(value)
+        except ValueError:
+            return (value, None)
```
Comment (pckroon) [inline bin/martinize2:285]: Needs a comment

**Review by pckroon — APPROVED** [review]
Some nitpicks, but also good enough as-is

**Review by pckroon — APPROVED** [review]

**Comment by jbarnoud** [comment]
> Address my own comments, add docstring for maxwarn function

Sorry, I forgot about the PR.

**Comment by pckroon** [comment]
Happens to the best of us :)

I'll leave it to you to press merge once travis is happy

**Pull request merged**